sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/transfer/TransferIdMappedToTransferInfo.java

```diff
@@ -0,0 +1,227 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.storage.blob.transfer;
+
+import android.content.Context;
+import android.util.Log;
+
+import androidx.annotation.MainThread;
+import androidx.annotation.NonNull;
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MediatorLiveData;
+import androidx.lifecycle.Transformations;
+import androidx.work.Data;
+import androidx.work.WorkInfo;
+import androidx.work.WorkManager;
+
+import java.util.List;
+
+/**
+ * Package private.
+ *
+ * A type to transform an input LiveData that emit a transfer id to an output LiveData that emit
+ * a stream of {@link TransferInfo}.
+ *
+ * Each {@link TransferInfo} event from output LiveData describes current state of the transfer
+ * identified by the transfer id.
+ */
+final class TransferIdMappedToTransferInfo {
```

Comment:
Just to clarify, the only reason we initialize this class with a `LiveData<long>` rather than just `long` is that we're using the LiveData as a communication channel, waiting for the transfer operation to be stored in the database so we can get its ID which happens in a callback. This implies that we only ever expect the `transferIdLiveData` to update once, right? 

If my understanding is correct we should document this somewhere. It wasn't obvious until I stared at this for awhile why we were using a `LiveData<long>` at all.

Reply:
you're right - we use `LiveData<Long>` as a communication channel, to receive the identifier that is not available at the time of creating `TransferIdMappedToTransferInfo` instance.

Regarding once vs multi emission - it emits only once in the current code base, but I have a plan to extend it for resume, after user pause, in such cases it can emit multiple times. Yet to be designed.

I will add reason for using  `LiveData<Long>` instead of long in the java-doc.